src/vs/base/browser/ui/menu/menu.ts

```diff
@@ -35,9 +49,31 @@ export class Menu {
 		menuContainer.setAttribute('role', 'presentation');
 		container.appendChild(menuContainer);
 
+		let parentData: ISubMenuData = {
+			parent: this
+		};
+
+		const getActionItem = (action: IAction) => {
+			if (action instanceof Separator) {
+				return new ActionItem(options.context, action, { icon: true });
+			} else if (action instanceof SubmenuAction) {
+				return new SubmenuActionItem(action, action.entries, parentData, options);
+			} else {
+				var menuItemOptions: IActionItemOptions = {};
```

Comment:
@sbatten in general we try to avoid `var` and use `let/const` (maybe check more code you added)

Reply:
@bpasero oops, sure thing